### DISCUSSION TITLE
How to not schedule a new run if a same job run is running

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8414] [Q&A] [answered]
hello dagster community
I have one quick question: I have job + scheduler to run this job every minute; is here some way to check is this job running now and do *not* start new one if it is (so just skip next schedule run)?
thank you in advance)

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1639478378159700?thread_ts=1639478378.159700&cid=C01U954MEER).*

**Comment by yuhan** [comment, marked as answer]
Setting max concurrent runs for the same job isn't supported out-of-box at the moment.

I'd suggest to write a custom logic in schedule to look up for runs of the same job in the run storage and then decide whether to kick off a new run, something like:

```python
from dagster import schedule, DagsterRunStatus, RunRequest
from dagster.core.storage.pipeline_run import RunsFilter


@job
def my_job():
    ...


@schedule(job=my_job, ...)
def my_schedule(context):
    # Find runs of the same job that are currently running
    run_records = context.instance.get_run_records(
        RunsFilter(job_name="my_job", statuses=[DagsterRunStatus.STARTED])
    )

    # Kick off a run only if no other runs of the same job are running
    if len(run_records) == 0:
        yield RunRequest(...)

```

**Reply by rrazavipour** [reply]
and how would that work for a Sensor as opposed to a scheduled job? DagsterRunStarted.STARTED also works with Sensors?

**Reply by javiermas** [reply]
It'd also be great to have something like this for automaterializations.
